packages/ui-text-input/src/TextInput/README.md

```diff
@@ -108,79 +185,131 @@ Focusable content will be focused separately from the input itself.
 > Note: For any content larger than an icon or small avatar (multiple [Tags](#Tag), for example),
 > use the `renderBeforeInput` property.
 
-```javascript
----
-type: example
----
-class ExtraContentExample extends React.Component {
-  constructor (props) {
-    super(props)
+- ```javascript
+  class ExtraContentExample extends React.Component {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        value: ''
+      }
+    }
 
-    this.state = {
-      value: ''
+    handleChange = (e, value) => this.setState({ value })
+
+    render() {
+      return (
+        <View as="div">
+          <TextInput
+            renderLabel="What are Paula Panda's favorite ice cream flavors?"
+            value={this.state.value}
+            onChange={this.handleChange}
+            renderBeforeInput={
+              <View display="block" padding="xxx-small 0">
+                {this.state.value !== '' && (
+                  <Tag
+                    text={this.state.value}
+                    margin="xxx-small xxx-small xxx-small none"
+                    onClick={function () {
+                      console.log(this.state.value)
+                    }}
```

Comment:
This throws an error in the console: Uncaught TypeError: Cannot read properties of undefined (reading 'state').
I think if you convert this to an arrow function (like you did in the function based example) it will work fine. Could you please fix?
Feel free to rewrite the original class-based component if you notice an error. :)

Reply:
Fixed this and i changed other anonymous functions to arrow functions to be more coherent